include/grpc/credentials.h

```diff
@@ -957,6 +957,25 @@ GRPCAPI void grpc_tls_credentials_options_set_min_tls_version(
 GRPCAPI void grpc_tls_credentials_options_set_max_tls_version(
     grpc_tls_credentials_options* options, grpc_tls_version max_tls_version);
 
+/**
+ * EXPERIMENTAL API - Subject to change
+ *
+ * Sets the identity ceritifcate provider in the options.
+ * The |options| will implicitly take a new ref to the |provider|.
+ */
+GRPCAPI void grpc_tls_credentials_options_set_identity_certificate_provider(
```

Comment:
Please mark the `grpc_tls_credentials_options_set_certificate_provider` as deprecated in favor of these two new APIs